src/transformers/models/ict/modeling_ict.py

```diff
@@ -0,0 +1,908 @@
+
+        output_attentions (`bool`, *optional*):
+            Whether or not to return the attentions tensors of all attention layers. See `attentions` under returned
+            tensors for more detail.
+        output_hidden_states (`bool`, *optional*):
+            Whether or not to return the hidden states of all layers. See `hidden_states` under returned tensors for
+            more detail.
+        interpolate_pos_encoding (`bool`, *optional*):
+            Whether to interpolate the pre-trained position encodings.
+        return_dict (`bool`, *optional*):
+            Whether or not to return a [`~utils.ModelOutput`] instead of a plain tuple.
+"""
+
+
+@add_start_docstrings(
+    "The ICT Model transformer outputting raw hidden-states without any specific head on top.",
+    ICT_TRANSFORMER_START_DOCSTRING,
+)
+# Copied from transformers.models.vit.modeling_vit.ViTModel with VIT->ICT,ViT->ICT
+class ICTTransformerModel(ICTTransformerPreTrainedModel):
+    def __init__(self, config: ICTTransformerConfig, use_mask_token: bool = False):
+        super().__init__(config)
+        self.config = config
+
+        self.token_embedding = nn.Embedding(config.vocab_size, config.hidden_size)
+        self.position_embedding = nn.Parameter(torch.zeros(1, config.block_size, config.hidden_size))
+        self.drop = nn.Dropout(config.residual_dropout_prob)
+
+        self.gradient_checkpointing = False
+        self.blocks = nn.ModuleList([ICTBlock(config) for _ in range(config.num_hidden_layers)])
+
+        # Decoder head
+        self.layernorm = nn.LayerNorm(config.hidden_size, eps=config.layer_norm_eps)
+        self.head = nn.Linear(config.hidden_size, config.vocab_size, bias=False)
+
+        # Initialize weights and apply final processing
+        self.post_init()
+
+    def get_input_embeddings(self):
+        return self.token_embedding
+
+    @add_start_docstrings_to_model_forward(ICT_TRANSFORMER_INPUTS_DOCSTRING)
+    @add_code_sample_docstrings(
+        checkpoint=_CHECKPOINT_FOR_DOC,
+        output_type=BaseModelOutput,
+        config_class=_CONFIG_FOR_DOC,
+        modality="vision",
+        expected_output=_EXPECTED_OUTPUT_SHAPE,
+    )
+    def forward(
+        self,
+        pixel_values: Optional[torch.Tensor] = None,
+        output_attentions: Optional[bool] = None,
+        output_hidden_states: Optional[bool] = None,
+        return_dict: Optional[bool] = None,
+    ) -> Union[Tuple, BaseModelOutput]:
+        all_hidden_states = () if output_hidden_states else None
+        all_self_attentions = () if output_attentions else None
+
+        if pixel_values is None:
+            raise ValueError("You have to specify pixel_values")
+
+        pixel_values = pixel_values.to(torch.long)
+        _, t = pixel_values.size()
+
+        inputs_embeds = self.token_embedding(pixel_values)
+
+        # if masks:
+        #     masks = masks.unsqueeze(2)
+        #     inputs_embeds = inputs_embeds * (1 - masks)
```

Comment:
I haven't yet added a feature to accept boolean masks @amyeroberts but I assume that should be a param to this forward method?

Reply:
Yes, exactly. You can take a look at `ViTForMaskedImageModeling`, `DeiTForMaskedImageModeling` and `SwinForMaskedImageModeling` classes to see the expected inputs